### PR TITLE
added live video conferencing to enable real-time interactions on stages

### PR DESCRIPTION
**Pull request by Deep-Codes**
This PR adds a live conferencing layer to the list of supported providers. Now developers can organize truly live virtual events using the template e.g - Webinars, Community Calls, Panel Discussions, etc. Any stage can now be converted into a live stage - with roles like speaker, moderator and attendee. Live stages also come with pre-built public chat section so that attendees and speakers can do seamless messaging while on the stage.

In case you do not require live stages, you can simply toggle the "isLive" radio button in the datocms dashboard and add a youtube URL instead.

We're actively working towards adding more features on the live audio-video side and are maintaining a public roadmap for everyone to keep track.

### CODE REVIEW

File: components/hms/EmptyRoom.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import Button from './Button';
+import { InviteIcon, PersonIcon } from '@100mslive/react-icons';
+import { selectLocalPeerRole } from '@100mslive/react-sdk';
+import { useHMSStore } from '@100mslive/react-sdk';
+import { ChangeRoleDialog } from './demo-cta/room-cta';
+import { useRouter } from 'next/router';
+
+const EmptyRoom = () => {
+  const role = useHMSStore(selectLocalPeerRole) || 'viewer';
+  const [cp, setCp] = React.useState(false);
```
Comment (leerob) [inline components/hms/EmptyRoom.tsx:12]: Could we have a better name here?

File: components/hms/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+      {dominantPeer ? (
+        <div className="md:flex  hidden items-center space-x-2">
+          <SpeakerIcon />
+          <span>{dominantPeer.name}</span>
+        </div>
+      ) : null}
+
+      <div className="flex flex-row justify-end gap-4">
+        {/* <div className="md:flex hidden flex-row  text-center align-middle gap-2">
```
Comment (leerob) [inline components/hms/Header.tsx:72]: Another

File: components/hms/SettingDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      </Button>
+      <audio
+        ref={audioRef}
+        src={TEST_AUDIO_URL}
+        onEnded={() => setPlaying(false)}
+        onPlay={() => setPlaying(true)}
+      ></audio>
+    </>
+  );
+};
```
Comment (leerob) [inline components/hms/SettingDialog.tsx:132]: Made it through to here, will revisit

**Review by leerob — COMMENTED** [review]
Before merging, we will want to run Prettier on the whole codebase

File: .eslintignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+postcss.config.js
```
[the remark this answers is not in the document]
Comment (Deep-Codes) [reply .eslintignore:1]: Yes

File: components/hms/EmptyRoom.tsx (hunk 1 above)
Comment (Deep-Codes) [reply components/hms/EmptyRoom.tsx:12]: Alright 👍🏽

**Comment by vercel-team** [comment]
#### 📝 Changed routes:


| [**/stage/\[slug\]**](https://conference-3c3fej8js.vercel.sh/stage/[slug]) |
|:-:|
|<a href="https://conference-3c3fej8js.vercel.sh/stage/[slug]">Screenshot of /stage/[slug]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/PQbbgSqC72dmqD7_wRYX_JIGUZzG8qSxNGwEhGv32p9bjNR0N-1xIzwgyeMeJQddGIhZ46n82EK-uIXdK1cLHA.png">(view full size)</a> |



And 3 other routes:
- [**/api/auth**](https://conference-3c3fej8js.vercel.sh/api/auth)
- [**/api/github-oauth**](https://conference-3c3fej8js.vercel.sh/api/github-oauth)
- [**/api/register**](https://conference-3c3fej8js.vercel.sh/api/register)



Commit fc734ca462b1b67dd58a45d515d90f203880c7f6 (https://conference-3c3fej8js.vercel.sh).

**Comment by leerob** [comment]
Okay, preview deployment is working now. However, it doesn't appear to be working. I can't view the live stage.

**Comment by Deep-Codes** [comment]
> Okay, preview deployment is working now. However, it doesn't appear to be working. I can't view the live stage.

The token endpoint does not have the trailing `/` 

NEXT_PUBLIC_HMS_TOKEN_ENDPOINT= `...app.100ms.live/`

the added one seems like

NEXT_PUBLIC_HMS_TOKEN_ENDPOINT= `...app.100ms.live`

**Comment by leerob** [comment]
Fixed!

File: lib/cms-providers/dato.ts [hunk 5]
```diff
@@ -13,136 +13,139 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { Job, Sponsor, Stage, Speaker } from '@lib/types';
+ import { Job, Sponsor, Stage, Speaker } from '@lib/types';
```
Comment (leerob) [inline lib/cms-providers/dato.ts:16]: Is Prettier set up here?

File: lib/redis.ts [hunk 6]
```diff
@@ -14,28 +14,29 @@
...
-    const result = hmac.digest('hex');
-    return result;
-  } else {
-    throw new Error('REDIS_EMAIL_TO_ID_SECRET is missing');
-  }
-}
-
-export default redis;
+ import Redis from 'ioredis';
+ import crypto from 'crypto';
```
Comment (leerob) [inline lib/redis.ts:18]: Looks like the formatting got all shifted by a little bit (back to Prettier comment)

File: styles/global.css [hunk 7]
```diff
@@ -18,13 +18,16 @@
   font-family: 'Inter';
   font-style: normal;
   font-weight: 100 900;
-  font-display: optional;
```
Comment (leerob) [inline styles/global.css:21]: Why did this change?

File: components/hms/ActiveSpeaker.tsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
+import {
```
Comment (leerob) [inline components/hms/ActiveSpeaker.tsx:1]: What is the folder of "hms" supposed to abbreviate?

**Review by leerob — APPROVED** [review]

**Pull request merged**